API.md

```diff
@@ -0,0 +1,41 @@
+# SegM8 API
+
+## `class SegM8`
+
+Creates an object of type `SegM8` to communicate with a chain of particular [SegM8 7-segment indicator modules](https://my.amperka.com/modules/SegM8).
+
+### `SegM8(uint8_t pinCS, uint8_t devicesCount = 1)`
+
+Constructs a new SegM8 object that uses the default hardware SPI bus.
+
+- `pinCS`: the chip select (also known as slave select) pin is used to control the shift-register latch. It is marked `CS` on the SegM8 board. Any GPIO pin of the controller may be used.
+- `deviceCount`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines a single module.
+
+### `SegM8(uint8_t pinCS, uint8_t pinMOSI, uint8_t pinSCK, uint8_t devicesCount = 1)`
+
+Constructs a new SegM8 object that uses a software SPI bus.
+
+- `pinCS`: the chip select (also known as slave select) pin is used to control the shift-register latch. It is marked `CS` on the SegM8 module. Any GPIO pin of the controller may be used.
+- `pinMOSI`: the MOSI (Master Out Slave In) pin is used to send data to the peripherals. It is marked `DI` on the SegM8 module. Any GPIO pin of the controller may be used.
+- `pinSCK`: the clock pin is used to clock pulses that synchronize data transmission generated by the master. It is marked `CLK` on the SegM8 module. Any GPIO pin of the controller may be used.
+- `deviceCount`: the number of SegM8 modules connected in a daisy-chain. If omitted, defines a single module.
+
+Difference between hardware and software SPI is interaction speed. Software SPI runs slower than hardware SPI but hardware SPI works only on specific board pins: `MOSI`, `MISO` (not used in SegM8) and `SCK`.
+
+### `void begin()`
+
+Initializes the given interface, prepares the board for the communication, and sets all module segments to the 'dark' state.
+Call this method before interacting with SegM8. For example, in the `setup()`.
+
+### `void printf(const char* format, ...)`
+
+Displays a 'c'-formatted string in a chain of `SegM8` indicator modules. The format string syntax matches the standard C `printf` function. For example: `Clock.printf("%2d:%2d",hours,minutes);`
+The 7-segment indicator output supports digits, excluding floats, and limited support of English characters: except for `K,M,N,V,W,X,Z`. Note that the display of letters on a 7-segment indicator does not always correspond to their generally accepted look.
+More supported characters: space, underscore, dot, minus, degree (at `^`). Unsupported characters are displayed as `blank space`.
+**Important: Arduino `xprintf` functions do not support `%f` format specifier correctly. Convert float/double in a string before printing the result.**
```

Comment:
No-no-no. No variadics in C++, please. It would be hell to maintain to work on all platforms correctly. Better offer several methods with clear intent:

```
void displayInt(int x, uint8_t flags);
void displayFloat(float x, int digits, uint8_t flags);
void displayString(const char* s, uint8_t flags);
```

where flags might be a combination of:

```
SegM8::ALIGN_LEFT
SegM8::ALIGN_RIGHT
SegM8::PAD_ZEROS
```

Reply:
done (some different)